mcm/tests/api_tools.py

```
@@ -0,0 +1,496 @@
+            requests.
+    """
+
+    def __init__(self, api: APIRequest, role: Roles):
+        self.api: APIRequest = api
+        self.role: Roles = role
+        self.logger = self.__logger()
+
+    def __logger(self) -> logging.Logger:
+        """
+        Returns a logger for the class
+        """
+        logger = logging.getLogger("mcm-rest-client")
+        formatter = logging.Formatter("[%(asctime)s][%(levelname)s] %(message)s")
+        handler = logging.StreamHandler(sys.stdout)
+
+        handler.setFormatter(formatter)
+        handler.setLevel(logging.INFO)
+        logger.addHandler(handler)
+        return logger
+
+    def _get(self, url) -> tuple[dict, Response]:
+        response = self.api.mcm_request(endpoint=url, method="GET", as_role=self.role)
+        return response.json(), response
+
+    def _put(self, url, data) -> tuple[dict, Response]:
+        response = self.api.mcm_request(
+            endpoint=url, method="PUT", data=data, as_role=self.role
+        )
+        return response.json(), response
+
+    def _delete(self, url) -> tuple[dict, Response]:
+        response = self.api.mcm_request(
+            endpoint=url, method="DELETE", as_role=self.role
+        )
+        return response.json(), response
+
+    # McM methods
+    def get(
+        self, object_type, object_id=None, query="", method="get", page=-1
+    ) -> dict | None:
+        """
+        Get data from McM
+        object_type - [chained_campaigns, chained_requests, campaigns, requests, flows, etc.]
+        object_id - usually prep id of desired object
+        query - query to be run in order to receive an object, e.g. tags=M17p1A, multiple parameters can be used with & tags=M17p1A&pwg=HIG
+        method - action to be performed, such as get, migrate or inspect
+        page - which page to be fetched. -1 means no paginantion, return all results
+        """
+        object_type = object_type.strip()
```

Remove the `strip()` and just let it fail. Providing it with spaces is a user error.

---

Agree. I will remove it